tests/test_integrals.py

```diff
@@ -688,3 +718,744 @@ def test_biest_interpolators(self):
+        np.testing.assert_allclose(z2, intersect[[0, 3, 5]], rtol=1e-6)
+
+    @pytest.mark.unit
+    def test_extrema_first_and_before_z2(self):
+        """Case where first intersect is extrema and second exits epigraph."""
+        # To make sure we do perform integral between first pair of intersects.
+        start = -1.2 * np.pi
+        end = -2 * start + 1
+        k = np.linspace(start, end, 7)
+        B = CubicHermiteSpline(
+            k,
+            np.cos(k) + 2 * np.sin(-2 * k) + k / 10,
+            -np.sin(k) - 4 * np.cos(-2 * k) + 1 / 10,
+        )
+        dB_dz = B.derivative()
+        pitch_inv = B(dB_dz.roots(extrapolate=False))[1] - 1e-13
+        z1, z2 = bounce_points(
+            pitch_inv, k, B.c.T, dB_dz.c.T, check=True, include_knots=True
+        )
+        z1, z2 = TestBounce1DPoints.filter(z1, z2)
+        assert z1.size and z2.size
+        # Our routine correctly detects intersection, while scipy, jnp.root fails.
+        intersect = B.solve(pitch_inv, extrapolate=False)
+        np.testing.assert_allclose(z1[0], -0.671904, rtol=1e-6)
+        np.testing.assert_allclose(z1, intersect[[0, 3, 5]], rtol=1e-5)
+        # intersect array could not resolve double root as single at index 0,1
+        np.testing.assert_allclose(intersect[0], intersect[1], rtol=1e-5)
+        np.testing.assert_allclose(z2, intersect[[2, 4, 6]], rtol=1e-5)
+
+    @pytest.mark.unit
+    def test_get_extrema(self):
+        """Test computation of extrema of |B|."""
+        start = -np.pi
+        end = -2 * start
+        k = np.linspace(start, end, 5)
+        B = CubicHermiteSpline(
+            k, np.cos(k) + 2 * np.sin(-2 * k), -np.sin(k) - 4 * np.cos(-2 * k)
+        )
+        dB_dz = B.derivative()
+        ext, B_ext = _get_extrema(k, B.c.T, dB_dz.c.T)
+        mask = ~np.isnan(ext)
+        ext, B_ext = ext[mask], B_ext[mask]
+        idx = np.argsort(ext)
+
+        ext_scipy = np.sort(dB_dz.roots(extrapolate=False))
+        B_ext_scipy = B(ext_scipy)
+        assert ext.size == ext_scipy.size
+        np.testing.assert_allclose(ext[idx], ext_scipy)
+        np.testing.assert_allclose(B_ext[idx], B_ext_scipy)
+
+
+def _mod_cheb_gauss(deg):
+    x, w = chebgauss(deg)
+    w /= chebweight(x)
+    return x, w
+
+
+def _mod_chebu_gauss(deg):
+    x, w = roots_chebyu(deg)
+    w *= chebweight(x)
+    return x, w
+
+
+class TestBounce1DQuadrature:
+    """Test bounce quadrature."""
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize(
+        "is_strong, quad, automorphism",
+        [
+            (True, tanh_sinh(40), None),
+            (True, leggauss(25), "default"),
+            (False, tanh_sinh(20), None),
+            (False, leggauss_lob(10), "default"),
+            # sin automorphism still helps out chebyshev quadrature
+            (True, _mod_cheb_gauss(30), "default"),
+            (False, _mod_chebu_gauss(10), "default"),
+        ],
+    )
+    def test_bounce_quadrature(self, is_strong, quad, automorphism):
```

Comment:
Can you add a reminder to future me in the docstring here to add the analytical test that converts incomplete elliptic integrals to complete ones using the Reciprocal Modulus transformation along with this [link](https://dlmf.nist.gov/19.7#E4) . It's not urgent and can be done after bounce is merged.